src/browser-driver-manager.js

```
@@ -1,259 +1,172 @@
+    );
+  }
 
-      if (chunk.toString().indexOf('browser-driver-manager error') !== -1) {
-        process.exitCode = 1;
-        reject();
-      }
-    });
-    child.on('close', () => {
-      if (returnValue) {
-        // output all logs except the last one
-        for (let i = 0; i < lines.length - 1; i++) {
-          process.stdout.write(lines[i]);
-        }
-        resolve(lines[lines.length - 1].toString());
-      } else {
-        resolve();
-      }
-    });
-  });
-}
+  const platform = detectBrowserPlatform();
+
+  if (!platform) {
+    throw new Error('Unable to detect browser platform');
+  }
+  // This will sync the browser and chromedriver versions
+  let chromeBuildId;
+  try {
+    chromeBuildId = await resolveBuildId(Browser.CHROME, platform, version);
+  } catch (e) {
+    throw new Error(e);
+  }
 
-/**
- * Install ChromeDriver from npm
- * @param {String} version - Version to install
- * @param {Boolean} verbose - If verbose logging is enabled
- */
-function installChromeDriver(version, verbose) {
-  return new Promise(resolve => {
-    const child = spawn('npm', [
-      'install',
-      '--no-save',
-      `chromedriver@${version}`
-    ]);
-    if (verbose) {
-      child.stdout.on('data', chunk => {
-        process.stdout.write(chunk);
-      });
+  const currentVersion = await getVersion();
+  if (currentVersion) {
+    if (currentVersion === chromeBuildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
     }
-    child.stderr.on('data', chunk => {
-      process.stderr.write(chunk);
-    });
-    child.on('close', () => {
-      resolve();
-    });
+    console.log(
+      `Chrome and Chromedriver versions ${currentVersion} are currently installed. Overwriting.`
+    );
+  }
+
+  const installedChrome = await installBrowser(
```

This looks like it can throw an error if passed a version it doesn't support. Example: `chrome@97`. We should catch the error and throw a better message if the error is 404, indicating that the provided version is not supported. It probably also could fail for other reasons so might need to also report a general error.